### ISSUE TITLE
ensure_first_param-check should not consider hashes

### ISSUE THREAD

**Issue by Raskil** [rodjek/puppet-lint#512]
Hi,

I noticed that ensure_first_param-check considers ensure parameters that are available within hashes and flags them as incorrect. Example:

```puppet
transition { 'transition tempfile':
    resource   => File[$tempfile],
    attributes => {
      ensure  => 'file',
      path    => $tempfile,
      backup  => false,
      content => template("some content")
    }
    ,
    prior_to   => Exec['myexec'],
  }

```

This is incorrectly noticed as no-ensure_first_param, because of the ensure attribute that is inside of the attributes hash. But the ensure is not related to the transition resource.

I've a workaround with control comments in place, but it would be better if puppet-lint does not check for ensure parameters inside of hashes.

Regards

Raskil

**Comment by rnelson0** [comment]
@Raskil This was fixed in #490. Can you verify that this is no longer a problem for you when you use puppet-lint from git HEAD instead of the 2.0.0 gem?

**Comment by Raskil** [comment]
@rnelson0 
Thank you for the heads up. Using HEAD indeed fixes the issue.

**Comment by rnelson0** [comment]
@Raskil Awesome. We hope to release 2.1.0 sometime soon-ish.
